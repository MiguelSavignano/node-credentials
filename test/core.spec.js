const core = require('../src/core');
require('./helpers/matchers');
const complexJSON = require('./examples/transformValues/complex.json');

describe('core', () => {
  let NODE_MASTER_KEY = '8aa93853b3ff01c5b5447529a9c33cb9';
  const SHORT_NODE_MASTER_KEY = '12345678';
  let credentials = { key: 'value' };
  let credentialsString = JSON.stringify(credentials);
  test('encrypt', async () => {
    const result = await core.encrypt(NODE_MASTER_KEY, credentialsString);
    expect(result).validEncrypted();
  });

  test('encrypt with any key length', async () => {
    const result = await core.encrypt(SHORT_NODE_MASTER_KEY, credentialsString);
    expect(result).validEncrypted();
  });

  test('decrypt', async () => {
    const [result, iv] = await core.decrypt(
      NODE_MASTER_KEY,
      'dwAhexc3PhUrX9i4gutpy6Hb8endKm7hMCQALPspYEc=--84X822lxzoPbO9Jh2knEGA=='
    );
    expect(JSON.parse(result)).toEqual(credentials);
  });

  test('decrypt  with any key length', async () => {
    const [result, iv] = await core.decrypt(
      SHORT_NODE_MASTER_KEY,
      'YhSWFEmk0OOG1qQDmjkEjg==--hfHcXE55MQ0bDOHho2SLag=='
    );
    expect(JSON.parse(result)).toEqual(credentials);
  });

  test('transformValues sync', () => {
    const result = core.transformValues(complexJSON, (value) => `${value} MOCK`);
    expect(result).toMatchSnapshot();
  });

  test('transformValues async', async () => {
    const result = await core.transformValues(complexJSON, async (value) => `${value} MOCK`);
    expect(result).toMatchSnapshot();
  });
});
