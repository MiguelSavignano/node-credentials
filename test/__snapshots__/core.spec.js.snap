// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core encryptYAML with merge keys 1`] = `
"source:
  &base
  a: \\"N9176c4qD6CSk7dH7vKbeA==--SlFF0O9iHgKpcds+/6nbEg==\\" # my comment 1
  b: \\"oiWdPL0jXuboSpQGuWlU0A==--SlFF0O9iHgKpcds+/6nbEg==\\"
  c:
    - ca: kIyJwEj3fkkmyBj6hZtYjw==--SlFF0O9iHgKpcds+/6nbEg==
    - cb: hijXas73wleqP2WbWNxCqQ==--SlFF0O9iHgKpcds+/6nbEg==
target:
  <<: *base
  d: \\"YojRsQ4e3F3kZNR+xnEdRg==--SlFF0O9iHgKpcds+/6nbEg==\\"
  e: null
"
`;
